cmd/rpcdaemon/commands/tg_block.go

```diff
@@ -0,0 +1,43 @@
+package commands
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ledgerwatch/turbo-geth/common"
+	"github.com/ledgerwatch/turbo-geth/core/rawdb"
+	"github.com/ledgerwatch/turbo-geth/core/types"
+	"github.com/ledgerwatch/turbo-geth/rpc"
+)
+
+// GetHeaderByNumber returns a block's header by number
+func (api *TgImpl) GetHeaderByNumber(ctx context.Context, number rpc.BlockNumber) (*types.Header, error) {
+	tx, err := api.dbReader.Begin(ctx)
```

Comment:
Can you please change here (and below) to `tx, err := api.db.Begin(ctx, nil, false)`. That open read-only transaction instead of writeable one. It does not matter for the remove KV, but it does matter with `--chaindata` option

Reply:
Got it, thank you.